dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIRequestSettings.cs

```diff
@@ -0,0 +1,160 @@
+    /// </summary>
+    [JsonPropertyName("stop_sequences")]
+    [JsonPropertyOrder(6)]
+    public IList<string> StopSequences { get; set; } = Array.Empty<string>();
+
+    /// <summary>
+    /// How many completions to generate for each prompt. Default is 1.
+    /// Note: Because this parameter generates many completions, it can quickly consume your token quota.
+    /// Use carefully and ensure that you have reasonable settings for max_tokens and stop.
+    /// </summary>
+    [JsonPropertyName("results_per_prompt")]
+    [JsonPropertyOrder(7)]
+    public int ResultsPerPrompt { get; set; } = 1;
+
+    /// <summary>
+    /// The system prompt to use when generating text completions using a chat model.
+    /// Defaults to "Assistant is a large language model."
+    /// </summary>
+    [JsonPropertyName("chat_system_prompt")]
+    [JsonPropertyOrder(8)]
+    public string ChatSystemPrompt { get; set; } = OpenAIRequestSettings.DefaultChatSystemPrompt;
+
+    /// <summary>
+    /// Modify the likelihood of specified tokens appearing in the completion.
+    /// </summary>
+    [JsonPropertyName("token_selection_biases")]
+    [JsonPropertyOrder(9)]
+    public IDictionary<int, int> TokenSelectionBiases { get; set; } = new Dictionary<int, int>();
+
+    /// <summary>
+    /// Default value for chat system property.
+    /// </summary>
+    internal static string DefaultChatSystemPrompt { get; } = "Assistant is a large language model.";
+
+    /// <summary>
+    /// Default max tokens for a chat completion
+    /// </summary>
+    internal static int DefaultTextMaxTokens { get; } = 256;
+
+    /// <summary>
+    /// Create a new settings object with the values from another settings object.
+    /// </summary>
+    /// <param name="requestSettings">Template configuration</param>
+    /// <param name="defaultMaxTokens">Default max tokens</param>
+    /// <returns>An instance of OpenAIRequestSettings</returns>
+    public static OpenAIRequestSettings FromRequestSettings(AIRequestSettings? requestSettings, int? defaultMaxTokens = null)
+    {
+        if (requestSettings is null)
+        {
+            return new OpenAIRequestSettings()
+            {
+                MaxTokens = defaultMaxTokens
+            };
+        }
+
+        OpenAIRequestSettings? openAIRequestSettings = null;
+
+        if (requestSettings.GetType() == typeof(OpenAIRequestSettings))
+        {
+            openAIRequestSettings = (OpenAIRequestSettings)requestSettings;
+        }
+        else
+        {
+            var json = JsonSerializer.Serialize(requestSettings);
+            openAIRequestSettings = JsonSerializer.Deserialize<OpenAIRequestSettings>(json, s_options);
+        }
+
+        if (openAIRequestSettings is null)
+        {
+            openAIRequestSettings = new OpenAIRequestSettings()
+            {
+                MaxTokens = defaultMaxTokens
+            };
+        }
+
+        return openAIRequestSettings;
```

Comment:
Using return first approach this can reduce complexity and size.

```suggestion
        if (requestSettings is OpenAIRequestSettings openAIRequestSettings)
        {
            return openAIRequestSettings;
        }
        
        var json = JsonSerializer.Serialize(requestSettings);
        var openAIRequestSettings = JsonSerializer.Deserialize<OpenAIRequestSettings>(json, s_options);
        
        if (openAIRequestSettings is not null) 
        {
            return openAIRequestSettings;
        }
        
        return new OpenAIRequestSettings()
        {
            MaxTokens = defaultMaxTokens
        };
```